src/scheduler/resv_info.c

```diff
@@ -186,6 +186,7 @@ query_reservations(server_info *sinfo, struct batch_status *resvs)
 		free_schd_error(err);
 		return NULL;
 	}
+	resresv_arr[0] = NULL;
```

Comment:
I don't think this is quite enough.  If I understand the issue properly, if query_resv() fails, we free the array.  We need to have it NULL terminated to do that.  This will handle the case if the first call to query_resv() returns NULL.  It doesn't handle the case if any other call to query_resv() returns NULL.  We don't NULL terminate the array until after the loop is complete.  You need to move the NULL termination inside the bottom of the loop.